src/components/Reports/ReportsTrafficSources/ReportsTrafficSources.js

```diff
@@ -90,7 +94,9 @@ const ReportsTrafficSources = function({ domainName, dateFrom, dependencies: { d
                           <S.Bar
                             primary
                             style={{
-                              width: (trafficSource.withEmail / trafficSource.quantity) * 100 + '%',
+                              width: trafficSource.quantity
+                                ? (trafficSource.withEmail / trafficSource.quantity) * 100 + '%'
```

Comment:
I've noticed you have always a division, how about making a function for example `SafeDivide(trafficSource.withEmail,trafficSource.quantity )` that returns the division result or zero if the divisor is zero? Then you can reuse and avoid having an inline if in the html. What do you think?

Reply:
Ready! good catch. thanks!